File: dwtddbc01/javascript/fujita_alone/stack/lib/stack.js
(function() {
    var Stack = function() {
	this._maxSize = 16;
	this._top = new this.Entry(undefined, undefined);
    };
    Stack.prototype = new function() {
	this.Entry = function(prev, value) {
	    //var e = new Object();
	    if (prev != undefined) {
		this.index = prev.index+1;
	    } else {
		this.index = 0;
	    }
	    this.value = value;
	    this.prev = prev;
	};
	this.isEmpty = function() {
	    return this._top.index == 0;
	};
	this.maxSize = function () {
	    return this._maxSize;
	};
	this.push = function (v) {
	    if (this._top.index == 16) {
		throw new RangeError('stack overflow maxSize is '+this._maxSize);
	    }
	    this._top = new this.Entry(this._top, v);
	    return v;
	};

	this.pop = function() {
	    if (this._top.index == 0) {
		throw new RangeError('stack is empty');
	    }

	    var e = this._top;
	    this._top = this._top.prev;
	    return e.value;
	};

	this.top = function() {
	    return this._top.value;
	};

	this.size = function() {
	    if (this._top == undefined) {
		return 0;
	    } else {
		return this._top.index;
	    }
	};
    };

    module.exports = Stack;
}).call(this);

File: dwtddbc01/ruby/sasaki_natsume/stack.rb
class Stack
  def initialize(maxsize = 16)
    @size = 0
    @maxsize = maxsize
    @stack = Array.new(@maxsize)
  end
  def empty?
    if @size == 0
      true
    else
        false
    end
  end
  def size
    @size
  end
  def push(value)
    raise RangeError if @size >= @maxsize
    @stack[@size] = value
    @size += 1
  end
  def pop
    @stack[@size] = nil
    @size -= 1
  end
  def top
    raise RangeError if @size == 0
    @stack[@size - 1]
  end
  def maxSize
    @maxsize 
  end
end
